diff --git a/kernel/sys.c b/kernel/sys.c
index b276a5e..898dbf1 100644
--- a/kernel/sys.c
+++ b/kernel/sys.c
@@ -1261,29 +1261,37 @@ static int override_release(char __user *release, size_t len)
 return ret;
 }
 
+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+extern struct static_key_false susfs_is_uname_spoof_buffer_set;
+extern void susfs_spoof_uname(struct new_utsname* tmp);
+#endif
 static int override_version(struct new_utsname __user *name)
 {
 #ifdef CONFIG_F2FS_REPORT_FAKE_KERNEL_VERSION
	int ret;

	if (strcmp(current->comm, "fsck.f2fs"))
		return 0;

	ret = copy_to_user(name->release, CONFIG_F2FS_FAKE_KERNEL_RELEASE,
			   strlen(CONFIG_F2FS_FAKE_KERNEL_RELEASE) + 1);
	if (ret)
		return ret;

	ret = copy_to_user(name->version, CONFIG_F2FS_FAKE_KERNEL_VERSION,
			   strlen(CONFIG_F2FS_FAKE_KERNEL_VERSION) + 1);

	return ret;
 #else
	return 0;
 #endif
 }

 SYSCALL_DEFINE1(newuname, struct new_utsname __user *, name)
 {
    struct new_utsname tmp;
    struct task_struct *t;
	bool is_netmgrd = false;

	down_read(&uts_sem);
	memcpy(&tmp, utsname(), sizeof(tmp));
+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+	if (static_branch_likely(&susfs_is_uname_spoof_buffer_set))
+		susfs_spoof_uname(&tmp);
+#endif	
	up_read(&uts_sem);

	rcu_read_lock();
	for_each_thread(current, t) {
